### PR TITLE
Check request expiry before emitting a new one

### PR DESCRIPTION
**Pull request by jakubuid**

### CODE REVIEW

File: protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/SessionRequestQueue.kt [hunk 1]
```diff
@@ -3,4 +3,4 @@ package com.walletconnect.sign.engine
 import com.walletconnect.sign.engine.model.EngineDO
 import java.util.LinkedList
 
-internal val sessionRequestsQueue: LinkedList<EngineDO.SessionRequestEvent> = LinkedList()
\ No newline at end of file
+internal val sessionRequestEvetnsQueue: LinkedList<EngineDO.SessionRequestEvent> = LinkedList()
```
Comment (Elyniss) [inline protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/SessionRequestQueue.kt:6]:
```suggestion
internal val sessionRequestEventsQueue: LinkedList<EngineDO.SessionRequestEvent> = LinkedList()
```

File: protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/requests/OnSessionRequestUseCase.kt [hunk 2]
```diff
@@ -81,13 +79,13 @@ internal class OnSessionRequestUseCase(
...
-                val event = if (sessionRequestsQueue.isEmpty()) {
+                val event = if (sessionRequestEvetnsQueue.isEmpty()) {
                     sessionRequestEvent
                 } else {
-                    sessionRequestsQueue.first()
+                    sessionRequestEvetnsQueue.find { event -> CoreValidator.isExpiryWithinBounds(event.request.expiry) } ?: sessionRequestEvent
                 }
 
-                sessionRequestsQueue.addLast(sessionRequestEvent)
+                sessionRequestEvetnsQueue.addLast(sessionRequestEvent)
```
Comment (Elyniss) [inline protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/requests/OnSessionRequestUseCase.kt:88]: Should we notify somehow that the session request expired?

Comment (jakubuid) [reply protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/requests/OnSessionRequestUseCase.kt:88]: We could log to the logcat

**Review by kacperoak — APPROVED** [review]

File: protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/domain/SignEngine.kt [hunk 3]
```diff
@@ -278,7 +277,7 @@ internal class SignEngine(
                             val verifyContext =
                                 verifyContextStorageRepository.get(sessionRequest.request.id) ?: VerifyContext(sessionRequest.request.id, String.Empty, Validation.UNKNOWN, String.Empty)
                             val sessionRequestEvent = EngineDO.SessionRequestEvent(sessionRequest, verifyContext.toEngineDO())
-                            sessionRequestsQueue.addLast(sessionRequestEvent)
+                            sessionRequestEvetnsQueue.addLast(sessionRequestEvent)
```
Comment (TalhaAli00) [inline protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/domain/SignEngine.kt:280]:
```suggestion
                            sessionRequestEventsQueue.addLast(sessionRequestEvent)
```

**Review by TalhaAli00 — APPROVED** [review]

**Pull request merged**